mule-extensions-api/src/main/java/org/mule/runtime/extension/api/connectivity/TransactionalConnection.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.connectivity;
+
+/**
+ * Connections which support transaction must implement this interface
+ * in order to start or join a current transaction.
+ *
+ * @since 1.0
+ */
+public interface TransactionalConnection extends Transactional
+{
+
+    /**
+     * @return whether the transaction is in XA mode
+     */
+    boolean isXA();
```

Comment:
I'm trying to get the extension code 100% separated from the actual tx infrastructure. That means I need a way to communicate to the runtime whether the transactions needs to be XA or not. This seems like a natural way to do this, specially since I found some examples of current components doing it like this (DB)